tests/test_main.py

```diff
@@ -4,8 +4,135 @@
 from py21cmemu import Emulator
 
 
-def test_basic_prediction():
+def test_prediction():
     """Simply test that we can make a prediction without erroring."""
     emu = Emulator(version="latest")
     theta = np.random.rand(9 * 5).reshape((5, 9))
-    emu.predict(theta)
+
+    theta, output, errors = emu.predict(theta)
+
+    # Test writing
+    output.write("test_writing")
+
+
+def test_properties():
+    """Test that the properties are loaded correctly."""
+    from py21cmemu.properties import emulator_properties as properties
+
+    properties.limits
+
+
+def test_inputs():
+    """Test that we perform parameter normalization properly."""
+    from py21cmemu import EmulatorInput
+    from py21cmemu.properties import emulator_properties as properties
+
+    limits = properties.limits.copy()
+    limits[7, :] *= 1000.0  # keV to eV
+
+    single_param = np.random.rand(9)
+    inp = EmulatorInput().make_param_array(single_param, normed=True)
```

Comment:
You should make a single instance of the `EmulatorInput()` class, say `inputs = EmulatorInput()`, then use it elsewhere.